Add note about `parse_command_line` function

Python-37 opened the pull request.

bdarnell commented:
I think it's a good idea to make a more explicit comment about this, but really it's related to the `final` parameter and applies to both `parse_config_file` and `parse_command_line` (and it's more than just logging - any user-defined command line flags using the `callback` option are also affected). Or it could go in the `note` block about logging at the module-level docstring. 

Personally I think it probably makes the most sense to add this to the module level instead of repeating it in both `parse` functions, but I know some people skip the module-level docs and just focus on the functions. What do you think? Would you have seen this in the module docs or does it need to be on the function?

One subtle thing here is that it's sometimes useful to call a non-final `parse_command_line`, set some additional options manually, then call a final `parse_command_line` with empty args. We haven't documented that pattern before because I think it's pretty rare, but I have seen it at least once. Would that have been useful to you?

Python-37 commented:
> I think it's a good idea to make a more explicit comment about this, but really it's related to the `final` parameter and applies to both `parse_config_file` and `parse_command_line` (and it's more than just logging - any user-defined command line flags using the `callback` option are also affected). Or it could go in the `note` block about logging at the module-level docstring.
> 
> Personally I think it probably makes the most sense to add this to the module level instead of repeating it in both `parse` functions, but I know some people skip the module-level docs and just focus on the functions. What do you think? Would you have seen this in the module docs or does it need to be on the function?
> 
> One subtle thing here is that it's sometimes useful to call a non-final `parse_command_line`, set some additional options manually, then call a final `parse_command_line` with empty args. We haven't documented that pattern before because I think it's pretty rare, but I have seen it at least once. Would that have been useful to you?

Thanks for your detailed replying, I've changed the comment, and moved to the module level. What do you think about this?

bdarnell commented:
Thanks!
